src/main/java/org/reaktivity/nukleus/kafka/internal/stream/KafkaCacheServerProduceFactory.java

```diff
@@ -1001,11 +1002,19 @@ private void doProduceInitialData(
                     final int initialPadding = fan.initialPadding;
                     final int initialBudget = fan.initialBudget;
                     final int reserved = Math.min(remaining + initialPadding, initialBudget);
+                    final int dirtyFlag = nextEntry.flags();
 
                     assert partitionOffset >= cursor.offset : String.format("%d >= %d", partitionOffset, cursor.offset);
 
+                    init:
                     if (reserved >= initialPadding)
                     {
+                        if (dirtyFlag != CACHE_ENTRY_FLAGS_COMPLETED)
```

Comment:
This assumes that the only two possible values for `flags` are `completed` and `dirty` so if a third flag is added, then the assumption will break and this logic will change.

Suggest changing the condition to `if ((flags & CACHE_ENTRY_FLAGS_DIRTY) != 0)`